lib/dynamoid/adapter_plugin/aws_sdk_v2.rb

```diff
@@ -539,12 +546,20 @@ def scan(table_name, scan_hash, select_opts = {})
 
         Enumerator.new do |y|
           # Batch loop, pulls multiple requests until done using the start_key
+          # by paging in by batch size rather than limit size
+          eval_count = 0
           loop do
             results = client.scan(request)
-
             results.data[:items].each { |row| y << result_item_to_hash(row) }
 
-            if((lk = results[:last_evaluated_key]) && batch)
+            # If limit is set then keep track and stop requesting once
+            # completed. This is eval_limit so we compare to scanned_count
+            # rather than resulting data
+            eval_count += results.scanned_count
```

Comment:
I was not sure about this, considering the code was changed from 'limit' to 'eval_limit' I assumed this change was meant to limit number of records DynamoDB scanned (similar to DynamoDB's definition of limit) rather than actual limit of the data returned. I can see it being fairly bad to do regular 'limit' if we end up having a scan since potentially scanning over all data. Which is why I implemented against 'scanned_count'.

On the flip side, I can see how if filters remove items then if you get an empty set, you are a bit stuck with not being able to query more, I think the TODO in that case would mean facilitating a way to return the Last Evaluated Key so could actually continue paginating through results. But not sure best way to do this with the general style that these queries may be written in. Maybe a class variable would work best to store it with thread safety considerations.